File: 0280-Wiggle Sort/0280-Wiggle Sort.cpp
class Solution {
public:
    void wiggleSort(vector<int>& nums) {
        if (nums.size() == 0)
        {
            return;
        }
        bool down = false;
        for (int i = 0; i < nums.size() - 1; ++i)
        {
            if ((down && nums[i] < nums[i + 1]) || (!down && nums[i] > nums[i + 1]))
            {
                swap(nums[i], nums[i + 1]);
            }
            down = !down;
        }
    }
};

File: 0201-0300/0280-Wiggle Sort/0280-Wiggle Sort.py
class Solution:
    def wiggleSort(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        down = False
        for i in range(len(nums) - 1):
            if (down and nums[i] < nums[i + 1]) or (not down and nums[i] > nums[i + 1]):
                nums[i], nums[i + 1] = nums[i + 1], nums[i]
            down = not down
